lib/segment/src/data_types/order_by.rs

```diff
@@ -0,0 +1,160 @@
+use common::types::PointOffsetType;
+use itertools::Either;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use validator::Validate;
+
+use crate::id_tracker::IdTrackerSS;
+use crate::types::{PointIdType, Range};
+
+pub const INTERNAL_KEY_OF_ORDER_BY_VALUE: &str = "____ordered_with____";
+
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Copy, Clone, Default)]
+#[serde(rename_all = "snake_case")]
+pub enum Direction {
+    #[default]
+    Asc,
+    Desc,
+}
+
+impl Direction {
+    pub fn as_range(&self, from: f64) -> Range {
+        match self {
+            Direction::Asc => Range {
+                gte: Some(from),
+                ..Default::default()
+            },
+            Direction::Desc => Range {
+                lte: Some(from),
+                ..Default::default()
+            },
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Validate, Clone, Default)]
+#[serde(deny_unknown_fields)]
```

Comment:
Would it be possible to avoid that?
One possible pitfal here is that we require offset point to be not deleted between iterations, which might sometimes be problematic.
So maybe instead we could allow user to define `value_offset` explicitly?

Reply:
Sure, it was something that couldn't convince myself, but exposing it makes a lot of sense

Reply:
Done